platforms/jvm/plugins-jvm-test-suite/src/main/java/org/gradle/api/testing/toolchains/internal/TestNGToolchainParameters.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.testing.toolchains.internal;
+
+import org.gradle.api.provider.Property;
+
+/**
+ * Parameters for configuring a TestNG test toolchain.
+ *
+ * @since 8.5
+ */
+public interface TestNGToolchainParameters extends JvmTestToolchainParameters {
+    /**
+     * The version of TestNG to use for compiling and executing tests.
+     */
+    Property<String> getVersion();
```

Comment:
For consistency with other toolchain parameters, would this be better named `getTestNgVersion`?

Reply:
There actually was some method to this madness, however weak.  For the JUnitPlatform parameters, there are two different versions that can be configured, the `platformVersion` and then the engine version (e.g. `jupiterVersion`) so those have unique names.  For TestNG and JUnit, there's only the one version so we just use `version`.  I don't feel strongly about this, though, just explaining that this _was_ intentional.

Given that, do you still think it's better to use `testNgVersion`?  I could be convinced either way.